test/support/utils_sycl.h

```diff
@@ -111,7 +111,15 @@ sycl::queue get_test_queue()
     return my_queue;
 }
 
-template <sycl::usm::alloc alloc_type, typename TestValueType, typename TestName>
+template <sycl::usm::alloc alloc_type>
+constexpr bool
+required_test_sycl_buffer()
+{
+    return alloc_type == sycl::usm::alloc::shared;
+}
+
+template <sycl::usm::alloc alloc_type, typename TestValueType, typename TestName,
+          bool TestSyclBuffer = required_test_sycl_buffer<alloc_type>()>
```

Comment:
Should we have the defaulted template parameter at both "levels"?
I think this overload is only called from within this file, so perhaps it should require explicit parameters.

Same for the 2, 3 buffer cases.

Reply:
Fixed